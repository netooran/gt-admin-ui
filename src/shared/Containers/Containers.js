import styled from 'styled-components';

export const ViewPort = styled.div`
  width: 100vw;
  height: 100vh;
`;

export const Center = styled.div`
  display: grid;
  place-items: center;
  width: 100%;
  height: 100%;
`;

export const Flex = styled.div`
  display: flex;
`;

export const Row = styled(Flex)`
  flex-direction: row;
`;

export const Column = styled(Flex)`
  flex-direction: column;
`;
