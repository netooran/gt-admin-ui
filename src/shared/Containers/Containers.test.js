import { render } from '@testing-library/react';
import { Center, Column, Flex, Row, ViewPort } from './Containers';

test('renders flex container', () => {
  const { container } = render(<Flex />);
  expect(container).toMatchSnapshot();
});

test('renders row', () => {
  const { container } = render(<Row />);
  expect(container).toMatchSnapshot();
});

test('renders column', () => {
  const { container } = render(<Column />);
  expect(container).toMatchSnapshot();
});

test('renders center', () => {
  const { container } = render(<Center />);
  expect(container).toMatchSnapshot();
});

test('renders viewport which fills the screen', () => {
  const { container } = render(<ViewPort />);
  expect(container).toMatchSnapshot();
});
