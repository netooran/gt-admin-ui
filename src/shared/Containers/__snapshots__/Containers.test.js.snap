// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders center 1`] = `
.c0 {
  display: grid;
  place-items: center;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`renders column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0 c1"
  />
</div>
`;

exports[`renders flex container 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`renders row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 c1"
  />
</div>
`;

exports[`renders viewport which fills the screen 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
}

<div>
  <div
    class="c0"
  />
</div>
`;
